sharktank/sharktank/utils/testing.py

```diff
@@ -108,6 +110,234 @@ def assertFileWritten(self, p: Path):
+            work_dir: The directory to save the results and intermediate files.
+                      If not provided, a temporary directory is used and will be deleted at the end of the run.
+            skip_decode: Whether to skip the decode stage. If True, the decode stage will not be run, and the decode results will not be compared.
+        """
+        # Note: Here to prevent circular imports
+        from sharktank.models.llm.llm import PagedLlmModelV1
+        from sharktank.utils.evaluate import pad_tokens
+        from sharktank.utils.load_llm import TorchGenerator
+        from sharktank.utils.export_artifacts import ExportArtifacts
+
+        work_dir = work_dir if work_dir else self._temp_dir
+
+        self.config = config
+        self.skip_decode = skip_decode
+
+        # NOTE: These paths must match those used by load_llm.py::Batch when it dumps the values
+        self.prefill_iree_logits_path = [work_dir / "prefill_iree_logits.npy"]
+        self.prefill_eager_logits_path = work_dir / "prefill_eager_logits.npy"
+        self.prefill_token_ids_path = work_dir / "prefill_token_ids.npy"
+        self.prefill_seq_lens_path = work_dir / "prefill_seq_lens.npy"
+        self.prefill_seq_block_ids_path = work_dir / "prefill_seq_block_ids.npy"
+
+        # NOTE: The _0 is required to match load_llm.py::Batch's syntax.
+        self.decode_iree_results_path = [work_dir / "decode_iree_logits.npy"]
+        self.decode_eager_logits_path = work_dir / "decode_eager_logits_0.npy"
+        self.decode_next_tokens_path = work_dir / "decode_next_tokens_0.npy"
+        self.decode_seq_lens_path = work_dir / "decode_seq_lens_0.npy"
+        self.decode_seq_block_ids_path = work_dir / "decode_seq_block_ids_0.npy"
+        self.decode_start_positions_path = work_dir / "decode_start_positions_0.npy"
+
+        self.prefill_cache_state_paths = []
+        self.decode_cache_state_paths = []
+        for i in range(
+            self.config.pipeline_parallelism_size * self.config.tensor_parallelism_size
+        ):
+            piece = f"_{i}" if i > 0 else ""
+            prefill_name = f"prefill_cache_state{piece}.npy"
+            decode_name = f"decode_cache_state_0{piece}.npy"  # _0 is required
+            self.prefill_cache_state_paths.append(work_dir / prefill_name)
+            self.decode_cache_state_paths.append(work_dir / decode_name)
+
+        self.dataset_path = work_dir / "parameters.irpa"
+
+        Dataset(root_theta=theta, properties=self.config.to_properties()).save(
+            path=self.dataset_path
+        )
+        self.exporter = ExportArtifacts.from_config(
+            self.config,
+            irpa_path=self.dataset_path,
+            iree_hip_target=self.iree_hip_target,
+            iree_hal_target_device=self.iree_hal_target_device,
+            hip_device_id=self.iree_device,
+            output_name=work_dir / "model",
+        )
+
+        self.config.device = torch.device("cuda:0")
```

Comment:
Did you intend for this to be hardcoded?
This would fail if you don't have this device?